python/hsfs/core/vector_server.py

```diff
@@ -264,13 +279,34 @@ def get_feature_vectors(self, entry, passed_features=[]):
             )
         )
 
-        return list(
+        # get column names for pandas dataframe
+        feature_names = list(
+            map(
+                lambda result_dict: self._generate_vector(result_dict)[1],
+                [batch_transformed[0]],
+            )
+        )[0]
+
+        vectors = list(
             map(
-                lambda result_dict: self._generate_vector(result_dict),
+                lambda result_dict: self._generate_vector(result_dict)[0],
                 batch_transformed,
             )
         )
 
+        if return_type is list or return_type is None:
```

Comment:
This comparision is not up-to-date anymore, is it? Should be typing.List

Reply:
By default I provide `None` because docs is not happy with `list`. I know we discussed `"list"` but then I ended up 1st checking if type is `str` and then if its equals to `"list"`

Reply:
But you do `if return_type is list`. I don't think that's the comparison you want here, looking at the types listed here https://github.com/logicalclocks/feature-store-api/pull/1065/files#diff-23c4c851037c735b9ed3cc92344dc3a8aad4380e5cc31122f12c326f493dffa9R404

Reply:
so by `typing.List` I though I am telling users they can use `list`.

Reply:
Ah okay, I understand now, but either way, it doesn't seem clean to let the user pass the 'list' module instead of a type.